sdk/core/azure-core/azure/core/pipeline/policies/_retry.py

```diff
@@ -455,7 +455,7 @@ def send(self, request):
                 # succeed--we'll never have a response to it, so propagate the exception
                 raise
             except AzureError as err:
-                if self._is_method_retryable(retry_settings, request.http_request):
+                if absolute_timeout > 0 and self._is_method_retryable(retry_settings, request.http_request):
```

Comment:
If absolute_timeout <= 0, we already raised in https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/core/azure-core/azure/core/pipeline/policies/_retry.py#L295?

Not true?

Reply:
True, but the problem is that we handle that error here by sleeping and iterating again.

Reply:
I might mis-understand, why absolute_timeout can be <= 0 here? 

Do you want to move

```py
end_time = time.time() 
if absolute_timeout:
    absolute_timeout -= (end_time - start_time)
```

here?

Reply:
This code is tricky to refactor without breaking something. For now we need to decrement absolute_timeout in the `finally` block so it's done after every iteration, including ones that end with the `continue` in the `try` block. I'm picking low hanging fruit in this PR. I think fixing everything (e.g. #18552) will require more thought and a larger change.

As for how absolute_timeout can be non-positive here, note that nothing breaks the loop when it is. _configure_timeout raises an exception when absolute_timeout <= 0 but this `except` block handles that exception because it's an AzureError. If the request is retryable, the `except` block sleeps and we go back to the `try` block, where `_configure_timeout` raises again. Repeat until retries are exhausted.